sql/src/test/java/io/crate/planner/WhereClauseOptimizerTest.java

```diff
@@ -0,0 +1,161 @@
+                new PartitionName("parted_pk", singletonList(new BytesRef("1395961200000"))).asIndexName(),
+                new PartitionName("parted_pk", singletonList(null)).asIndexName()
+            )
+            .build();
+    }
+
+    private WhereClauseOptimizer.DetailedQuery optimize(String statement) {
+        QueriedTable queriedTable = e.analyze(statement);
+        EvaluatingNormalizer normalizer = new EvaluatingNormalizer(
+            e.functions(),
+            RowGranularity.CLUSTER,
+            null,
+            queriedTable.tableRelation()
+        );
+        return WhereClauseOptimizer.optimize(
+            normalizer,
+            queriedTable.where().queryOrFallback(),
+            ((DocTableInfo) queriedTable.tableRelation().tableInfo()),
+            e.getPlannerContext(clusterService.state()).transactionContext()
+        );
+    }
+
+    @Test
+    public void testFilterOn_IdDoesNotProduceDocKeysWithOnlyClusteredBy() {
+        WhereClauseOptimizer.DetailedQuery query = optimize(
+            "select * from tclustered where _id = '1'");
+        assertThat(query.docKeys().isPresent(), is(false));
+    }
+
+    @Test
+    public void testFilterOn_IdOnPartitionedTableDoesNotResultInDocKeys() {
+        WhereClauseOptimizer.DetailedQuery query = optimize(
+            "select * from parted where _id = '1'");
+        assertThat(query.docKeys().isPresent(), is(false));
+    }
+
+    @Test
+    public void testFilterOnPartitionColumnAndPrimaryKeyResultsInDocKeys() {
+        WhereClauseOptimizer.DetailedQuery query = optimize(
+            "select * from parted_pk where id = 1 and date = 1395874800000");
+        assertThat(query.docKeys().toString(), is("Optional[DocKeys{1, 1395874800000}]"));
+        assertThat(query.partitions(), empty());
+    }
+
+    @Test
+    public void testClusteredByValueContainsComma() throws Exception {
+        WhereClauseOptimizer.DetailedQuery query = optimize(
+            "select * from bystring where name = 'a,b,c'");
+        assertThat(query.clusteredBy(), contains(isLiteral("a,b,c")));
+        assertThat(query.docKeys().get().size(), is(1));
+        assertThat(query.docKeys().get().getOnlyKey(), isDocKey("a,b,c"));
+    }
+
+    @Test
+    public void testEmptyClusteredByValue() throws Exception {
+        WhereClauseOptimizer.DetailedQuery query = optimize("select * from bystring where name = ''");
+        assertThat(query.clusteredBy(), contains(isLiteral("")));
+        assertThat(query.docKeys().get().getOnlyKey(), isDocKey(""));
+    }
+
+    @Test
+    public void testFilterOnClusteredByColumnDoesNotResultInDocKeys() {
+        WhereClauseOptimizer.DetailedQuery query = optimize(
+            "select * from clustered_by_only where x = 10");
+        assertThat(query.clusteredBy(), contains(isLiteral(10)));
+        assertThat(query.docKeys().isPresent(), is(false));
+    }
+
+    @Test
+    public void testFilterOnClusteredByColumnDoesNotResultInDocKeysSimpleEq() {
```

Comment:
what is the difference to `testFilterOnClusteredByColumnDoesNotResultInDocKeys` ? looks the same